lib/drivers/raicoin.js

```diff
@@ -0,0 +1,83 @@
+const Driver = require('../models/driver');
+const Supply = require('../models/supply');
+const { promisesMap } = require('../util');
+
+/**
+ * Raicoin driver. Supports total supply
+ * and account balance for RAI coin.
+ *
+ * @augments Driver
+ * @memberof Driver
+ */
+class Raicoin extends Driver {
+  constructor(options) {
+    super({
+      timeout: 200, // 5 requests per second
+      supports: {
+        balances: true,
+      },
+      options,
+    });
+  }
+
+  /**
+   * @augments Driver.fetchTotalSupply
+   * @async
+   */
+  async fetchTotalSupply() {
+    const supply = await this.request('https://rpc.raicoin.org/supply');
+    return Number(supply);
+  }
+
+  /** get balance for specific wallet address
+   *
+   * @augments Driver.fetchBalance
+   * @param {SupplyModifier} modifier {@link SupplyModifier}
+   * @async
+   */
+  async fetchBalance(modifier) {
+    const { head_block: { balance } } = await this.request({
+      url: 'https://rpc.raicoin.org',
+      body: {
+        account: modifier,
+        action: 'account_info',
+      },
+      method: 'POST',
+    });
+
+    return Number(balance) / 10 ** 9;
```

Comment:
Are you sure this is the correct decimal correction? That leaves only 32 Raicoin in balance on the account you use as modifier, and according to coinranking Raicoin is not worth a lot per unit....

Reply:
@wouthoekstra  I think they have a different representation of balance amount on their official site, because account's balance can not be greater then the total supply, right? Also, if you check this account https://raicoin.org/explorer.html?search=rai_1b5wb8hs5d3u5q8cnesk5xprr4damryn9xmrju7grdrtq6sxpanrwjo4s4r3 and see the table's fourth column 'weight', it's actually represents a balance of the specific account address :) so the conclusion is that they use 10^9 decimals.